src/topology/algebra/ring.lean

```diff
@@ -129,41 +147,39 @@ begin
   exact hadd.comp (((hmul_right y₀).comp tendsto_fst).prod_mk ((hmul_left  x₀).comp tendsto_snd))
 end
 
-lemma topological_ring.of_nhds_zero
+lemma topological_semiring.of_nhds_zero
   (hadd : tendsto (uncurry ((+) : R → R → R)) ((𝓝 0) ×ᶠ (𝓝 0)) $ 𝓝 0)
   (hneg : tendsto (λ x, -x : R → R) (𝓝 0) (𝓝 0))
   (hmul : tendsto (uncurry ((*) : R → R → R)) ((𝓝 0) ×ᶠ (𝓝 0)) $ 𝓝 0)
   (hmul_left : ∀ (x₀ : R), tendsto (λ x : R, x₀ * x) (𝓝 0) $ 𝓝 0)
   (hmul_right : ∀ (x₀ : R), tendsto (λ x : R, x * x₀) (𝓝 0) $ 𝓝 0)
-  (hleft : ∀ x₀ : R, 𝓝 x₀ = map (λ x, x₀ + x) (𝓝 0)) : topological_ring R :=
+  (hleft : ∀ x₀ : R, 𝓝 x₀ = map (λ x, x₀ + x) (𝓝 0)) : topological_semiring R :=
 begin
   haveI := topological_add_group.of_comm_of_nhds_zero hadd hneg hleft,
-  exact topological_ring.of_add_group_of_nhds_zero hmul hmul_left hmul_right
+  exact topological_semiring.of_add_group_of_nhds_zero hmul hmul_left hmul_right
 end
 
 end
 
-variables {α} [ring α] [topological_space α] [topological_ring α]
+variables {α} [ring α] [topological_space α] [topological_semiring α]
 
 @[priority 100] -- See note [lower instance priority]
-instance topological_ring.to_topological_add_group : topological_add_group α :=
-{ continuous_add := continuous_add,
-  continuous_neg := by simpa only [neg_one_mul, id.def] using
-    (@continuous_const α α _ _ (-1)).mul continuous_id }
+instance topological_semiring.to_topological_add_group : topological_add_group α :=
+{ ..(topological_semiring.has_continuous_neg_of_mul α) }
```

Comment:
I'd be tempted to remove this, and only have `topological_ring.to_topological_add_group`.

Am I right in thinking that `[ring α] [topological_semiring α]` and `[ring] [topological_ring α]` hold mathematically the same content?

Reply:
Yes, that's how we've gotten by without `has_continuous_neg` for so long. The reason I left it this way is because everything in mathlib is currently phrased in terms of the (new) `topological_semiring`, so moving this into `topological_ring` would break stuff. Let me know what you think.

Reply:
I have tried to uniformize across mathlib so that whenever there is a `[ring α]` assumption, we use `[topological_ring α]` instead of `[topological_semiring α]`. I have also provided a constructor `topological_semiring.to_topological_ring` which allows the user to create a topological ring by just verifying the conditions for a topological semiring (assuming an instance of `[non_assoc_ring α]`).